twister: add `--coverage-formats` option for use with gcovr

jpwright opened the pull request:
Adds an `--coverage-formats` option to twister, which (if gcovr is used as the coverage tool) allows generation of alternate report formats such as coberatura xml, gcc txt, etc.

Closes #46580.

Commit:
scripts: add --coverage-formats option for use with gcovr

Adds optional twister flag to support gcovr output report formats.

Signed-off-by: Jason Wright <jwright@synchron.com>

nashif commented on scripts/pylib/twister/twisterlib/enviornment.py:
```diff
@@ -247,6 +247,11 @@ def parse_arguments(args):
     parser.add_argument("--coverage-tool", choices=['lcov', 'gcovr'], default='lcov',
                         help="Tool to use to generate coverage report.")
 
+    parser.add_argument("--coverage-formats", action="store", default='html',
```
maybe you should make this option depend on coverage tool selected being gcovr?

Commit:
scripts: restrict --coverage-formats option to gcovr

Generates an error if --coverage-formats option is supplied without
coverage tool being set to gcovr.

Signed-off-by: Jason Wright <jwright@synchron.com>

enjiamai commented:
Just a minor issue: we should also make some change in the report output to adapt this:
In scripts/pylib/twister/twisterlib/coverage.py
```
101                 logger.info("HTML report generated: {}".format(
102                     os.path.join(outdir, "coverage", "index.html")))
```
otherwise the report will always show:
INFO    - HTML report generated: /home/mm/zephyrproject/zephyr/twister-out/coverage/index.html
if  you use csv, txt or other format.

Commit:
scripts: better logging for --coverage-formats option

Adds additional log messages for coverage reports
beyond HTML format.

Signed-off-by: Jason Wright <jwright@synchron.com>

jpwright commented:
thank you @enjiamai - @nashif could you please re-approve?
